### PR TITLE
Makes fire alarm simpler

### PR DESCRIPTION
**Pull request by nicbn**
[Changelogs]: #
:cl: nicbn
tweak: Fire alarm is now simpler. Touch it to activate, touch it to deactivate. When activated, it will blink inconsistently if it is emagged.
/:cl:

Fire alarms functionality is too simple to deserve a tgui. This makes toggling faster, also it feels better to interact with it as there is no window popping into your screen.

A lot of air alarms were inverted in the maps, so the fire icon that replaces the old lights were inverted as well. I fixed this.

The icon now has two lights, one that shows status alert and one that blinks red when the alarm is ringing. The empty space was covered with a grey fire outline that will glow in the dark.

### CODE REVIEW

**Comment by GunHog** [comment]
Against this because:
1) There is no other public indicator of station alert status for those that missed the announcements
2) The emag change makes emagging / malf hacking them less viable. The current system requires someone to check it, rather than anyone in eyeshot of it being alerted.
3) You could add a mouse shortcut (such as ctrl-click) for toggling the alarm faster, if your goal is simply making the alarm usable without loading the UI (which might take 10+ seconds or never load)

**Comment by nicbn** [comment]
Ok I will readd the alert status and make it so that the only way to tell if an alarm is emagged is by activating it.

**Comment by nicbn** [comment]
No it only toggle with empty hands
The proc is literally called `attack_hand`

**Comment by deathride58** [comment]
Fire alarms have a 33% chance of toggling on/off when damaged by *anything.* Projectiles, toolboxes, etc, are all capable of toggling fire alarms.

Over at Citadel, we made it so that right-clicking fire alarms quickly toggles them if possible. It might be worth taking inspiration from that

**Comment by nicbn** [comment]
Hmm maybe in another PR

**Comment by GunHog** [comment]
Please make sure that the AI can sill use it as well. I did not find an attack_ai() in your code. This was previously handled by the TGUI.

**Comment by nicbn** [comment]
Gimme some time

**Comment by imsxz** [comment]
freddy alarms buff

**Comment by Tlaltecuhtli** [comment]
remove tgui

**Comment by optimumtact** [comment]
please update your OP to reflect the current state of this PR

**Comment by nicbn** [comment]
It already is up to date

File: code/game/machinery/firealarm.dm [hunk 1]
```diff
@@ -69,23 +69,22 @@
...
+		add_overlay("fire_[SEC_LEVEL_GREEN]")
+		SSvis_overlays.add_vis_overlay(src, icon, "fire_[SEC_LEVEL_GREEN]", ABOVE_LIGHTING_LAYER, ABOVE_LIGHTING_PLANE, dir)
 
-	if(detecting)
-		add_overlay("overlay_[A.fire ? "fire" : "clear"]")
-		SSvis_overlays.add_vis_overlay(src, icon, "overlay_[A.fire ? "fire" : "clear"]", ABOVE_LIGHTING_LAYER, ABOVE_LIGHTING_PLANE, dir)
-	else
-		add_overlay("overlay_fire")
-		SSvis_overlays.add_vis_overlay(src, icon, "overlay_fire", ABOVE_LIGHTING_LAYER, ABOVE_LIGHTING_PLANE, dir)
+	add_overlay("fire_[detecting && A.fire ? (obj_flags & EMAGGED) ? "emagged" : "on" : "off"]")
```
Comment (AnturK) [inline code/game/machinery/firealarm.dm]: This icon state name is complicated enough to put into variable.

**Review by AnturK — CHANGES_REQUESTED** [review]
Just this and this is good to go.

**Review by AnturK — APPROVED** [review]

**Pull request merged**